contrib/vcl/source/vcl_io_handle.h

```diff
@@ -0,0 +1,143 @@
+#pragma once
+
+#include <list>
+
+#include "envoy/api/io_error.h"
+#include "envoy/network/io_handle.h"
+
+#include "source/common/common/logger.h"
+#include "source/common/network/io_socket_error_impl.h"
+
+#include "vppcom.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Network {
+namespace Vcl {
+
+using namespace Envoy::Network;
+
+#define VCL_INVALID_SH uint32_t(~0)
+#define VCL_SH_VALID(_sh) (_sh != static_cast<uint32_t>(~0))
+#define VCL_SET_SH_INVALID(_sh) (_sh = static_cast<uint32_t>(~0))
+
+int peekVclSession(vcl_session_handle_t sh, vppcom_endpt_t* ep, uint32_t* proto);
+Envoy::Network::Address::InstanceConstSharedPtr vclEndptToAddress(const vppcom_endpt_t& endpt,
+                                                                  uint32_t sh);
+
+class VclIoHandle : public Envoy::Network::IoHandle, Logger::Loggable<Logger::Id::connection> {
+public:
+  explicit VclIoHandle(os_fd_t fd = INVALID_SOCKET) : sh_(fd) {
```

Comment:
Does it mean that `sh_` and `fd_` is the same thing?

Reply:
Probably, you can use the `NonCopyable` mixin class to check it at compile time.

Reply:
It's meant more as a means to discover the issue when it happens and to fix it, as opposed to breaking the build for others while adding features.

Reply:
Actually, apologies, I'm at fault here for not really checking recent `IoHandle` changes! Added the `NonCopyable` mixin class and removed the constructor.